### PR TITLE
Allow Impersonation From Admin Panel

### PR DESCRIPTION
**Pull request by DavideIadeluca**
**Fixes https://discuss.flarum.org/d/9868-fof-impersonate-login-as-other-users/63**

**Changes proposed in this pull request:**
- Create Reusable Button to login as other user
- Move components to `common`
- Extend `UserListPage` with new Button
- Set flarum/core constraint to ^1.8.6

**Reviewers should focus on:**
Focus especially on how I've implemented the routing from the admin frontend to the user profile page. Using `app.route.user(user)` is not possible from the backend so I had to use a somewhat hacky solution to achieve that. Is there perhaps a way how we can do this better?

**Screenshot**
Screenshot 2024-10-03 at 07 10 54

**Confirmed**

- [ ] Frontend changes: tested on a local Flarum installation.
- [ ] Backend changes: tests are green (run `composer test`).

**Required changes:**

- [ ] Related [Flarum core extension PR's](https://github.com/flarum/core/pulls): (Omit this section if irrelevant)

### CODE REVIEW

File: composer.json [hunk 1]
```diff
@@ -34,7 +34,7 @@
         }
     ],
     "require": {
-        "flarum/core": "^1.2.0"
+        "flarum/core": "^1.8.7"
```
Comment (imorland) [inline composer.json:37]:
```suggestion
        "flarum/core": "^1.8.6"
```
Although 1.8.6 is buggy with the basicspage, technically this is the minimum supported version for these changes

**Review by imorland — COMMENTED** [review]
Other than the constraint on core, all good here.

I'd also like to see `fof/moderator-notes` updated to support the admin `LogInAsUser` functionality as well, please. It's quite an important integration IMO..

File: composer.json (hunk 1 above)
Comment (DavideIadeluca) [reply composer.json:37]: Yeah I agree, but the CI won't pass if we use 1.8.6, see https://github.com/FriendsOfFlarum/impersonate/actions/runs/11156322074/job/31450506821

Comment (imorland) [reply composer.json:37]: How random! Ok, no worries then :)

**Comment by DavideIadeluca** [comment]
> I'd also like to see `fof/moderator-notes` updated to support the admin `LogInAsUser` functionality as well, please. It's quite an important integration IMO..

Can you please elaborate here? Creating moderator notes from the admin user list works already:

Screenshot 2024-10-21 at 07 42 47

**Comment by imorland** [comment]
My mistake, on checking the integration is based on a backend listener, so nothing to do here at all 🙈

**Review by imorland — APPROVED** [review]

**Pull request merged**